### PR TITLE
Refactor branch 2.5, clean code

### PR DESCRIPTION
**Pull request by bcode2**
This PR...

Fixes issue #

## Changes
refactor: replace while loop with enhanced for loop
refactor: language migration; simplify maps operations
style: improve visibility change l by L for long literals
refactor: replace addAll() with  parameterized constructor
refactor: set serialVersionUID  fields as static and final
refactor: language migration, remove unnecessary boxing
refactor: language migration, use lambda expression to simplify code and remove unused imports
refactor: remove redundant variable declaration
performance: language migration
refactor: simplify boolean expression

-----------------
## Checklist
- [x] tested locally
- [ ] updated the docs
- [ ] added appropriate test
- [x] signed-off on the DCO referenced in the CONTRIBUTING link below via `git commit -s` on my commits, and submit this code under terms of the Apache 2.0 license and assign copyright to the Quartz project owners
  (If you're not using command-line, you can use a [browser extension](https://github.com/scottrigby/dco-gh-ui) )
-----------------
In submitting this contribution, I agree to the terms of contributing as referred to here: 
https://github.com/quartz-scheduler/contributing/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by melloware — APPROVED** [review]
great work @bcode2

File: quartz/src/main/java/org/quartz/impl/jdbcjobstore/StdJDBCDelegate.java [hunk 1]
```diff
@@ -3118,10 +3110,9 @@ protected Properties convertToProperty(Map<?, ?> data) throws IOException {
         Properties properties = new Properties();
 
         for (Map.Entry<?, ?> value : data.entrySet()) {
-            Map.Entry<?, ?> entry = (Map.Entry<?, ?>) value;
 
-            Object key = entry.getKey();
-            Object val = (entry.getValue() == null) ? "" : entry.getValue();
+            Object key = ((Map.Entry<?, ?>) value).getKey();
```
Comment (chrisdennis) [inline quartz/src/main/java/org/quartz/impl/jdbcjobstore/StdJDBCDelegate.java:3114]: Unusual casts here? Needed?

File: quartz/src/main/java/org/quartz/impl/triggers/DailyTimeIntervalTriggerImpl.java [hunk 2]
```diff
@@ -875,8 +875,6 @@ public Set<Integer> getDaysOfWeek() {
     public void setDaysOfWeek(Set<Integer> daysOfWeek) {
         if(daysOfWeek == null || daysOfWeek.isEmpty())
             throw new IllegalArgumentException("DaysOfWeek set must be a set that contains at least one day.");
-        else if(daysOfWeek.isEmpty())
```
Comment (chrisdennis) [inline quartz/src/main/java/org/quartz/impl/triggers/DailyTimeIntervalTriggerImpl.java:878]: Do we want to fix this by changing the previous condition and modifying the messages?

File: quartz/src/main/java/org/quartz/plugins/xml/XMLSchedulingDataProcessorPlugin.java [hunk 3]
```diff
@@ -214,32 +213,29 @@ public void start(UserTransaction userTransaction) {
...
-                        
+
                         // remove preexisting job/trigger, if any
                         getScheduler().unscheduleJob(tKey);
 
                         JobDetail job = newJob().withIdentity(jobTriggerName, JOB_INITIALIZATION_PLUGIN_NAME).ofType(FileScanJob.class)
-                            .usingJobData(FileScanJob.FILE_NAME, jobFile.getFileName())
-                            .usingJobData(FileScanJob.FILE_SCAN_LISTENER_NAME, JOB_INITIALIZATION_PLUGIN_NAME + '_' + getName())
-                            .build();
+                                .usingJobData(FileScanJob.FILE_NAME, jobFile.getFileName())
```
Comment (chrisdennis) [inline quartz/src/main/java/org/quartz/plugins/xml/XMLSchedulingDataProcessorPlugin.java:226]: Not a huge fan of this kind of reformatting in a such a large PR.

File: quartz/src/main/java/org/quartz/simpl/CascadingClassLoadHelper.java [hunk 4]
```diff
@@ -107,9 +106,8 @@ public Class<?> loadClass(String name) throws ClassNotFoundException {
         Class<?> clazz = null;
         ClassLoadHelper loadHelper = null;
 
-        Iterator<ClassLoadHelper> iter = loadHelpers.iterator();
-        while (iter.hasNext()) {
-            loadHelper = iter.next();
+        for (ClassLoadHelper helper : loadHelpers) {
+            loadHelper = helper;
```
Comment (chrisdennis) [inline quartz/src/main/java/org/quartz/simpl/CascadingClassLoadHelper.java:110]: Seems cleaner to do this assignment right before the `break`

File: quartz/src/main/java/org/quartz/xml/ValidationException.java [hunk 5]
```diff
@@ -127,10 +126,6 @@ public String getMessage() {
         boolean first = true;
 
         for (Exception e : getValidationExceptions()) {
-            if (!first) {
-                sb.append('\n');
-                first = false;
```
Comment (chrisdennis) [inline quartz/src/main/java/org/quartz/xml/ValidationException.java:132]: This looks to be a bug where this line belongs in an else.

File: quartz/src/main/java/org/quartz/core/jmx/JobDataMapSupport.java [hunk 6]
```diff
@@ -79,9 +78,7 @@ public static CompositeData toCompositeData(String key, String value) {
     public static TabularData toTabularData(JobDataMap jobDataMap) {
         TabularData tData = new TabularDataSupport(TABULAR_TYPE);
         ArrayList<CompositeData> list = new ArrayList<>();
-        Iterator<String> iter = jobDataMap.keySet().iterator();
-        while (iter.hasNext()) {
-            String key = iter.next();
+        for (String key : jobDataMap.keySet()) {
```
[the remark this answers is not in the document]
Comment (bcode2) [reply quartz/src/main/java/org/quartz/core/jmx/JobDataMapSupport.java:81]: yes, you right even my beloved  Sonarqube is complaining. I was  fixing simple issues first before going to the complex ones  :)  and grouping them in categories. Will fix them.... :)

File: quartz/src/main/java/org/quartz/impl/jdbcjobstore/StdJDBCDelegate.java [hunk 7]
```diff
@@ -3077,11 +3070,10 @@ protected ByteArrayOutputStream serializeJobData(JobDataMap data)
      */
     protected Object getKeyOfNonSerializableValue(Map<?, ?> data) {
         for (Map.Entry<?, ?> value : data.entrySet()) {
-            Map.Entry<?, ?> entry = (Map.Entry<?, ?>) value;
 
-            try (ByteArrayOutputStream baos = serializeObject(entry.getValue())) {
+            try (ByteArrayOutputStream baos = serializeObject(((Map.Entry<?, ?>) value).getValue())) {
```
[the remark this answers is not in the document]
Comment (bcode2) [reply quartz/src/main/java/org/quartz/impl/jdbcjobstore/StdJDBCDelegate.java:3074]: umm I would prefer try-with-resources. Much safer

File: quartz/src/main/java/org/quartz/impl/triggers/DailyTimeIntervalTriggerImpl.java (hunk 2 above)
Comment (bcode2) [reply quartz/src/main/java/org/quartz/impl/triggers/DailyTimeIntervalTriggerImpl.java:878]: for me it is redundant both messages. The fact would be to choose which one will be left :). There is a pr opened for this :) https://github.com/quartz-scheduler/quartz/issues/1209

File: quartz/src/main/java/org/quartz/plugins/xml/XMLSchedulingDataProcessorPlugin.java (hunk 3 above)
Comment (bcode2) [reply quartz/src/main/java/org/quartz/plugins/xml/XMLSchedulingDataProcessorPlugin.java:226]: Yes  could be right, It is just a matter of preference. Do you want me to revert it?,

File: quartz/src/main/java/org/quartz/impl/jdbcjobstore/StdJDBCDelegate.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (chrisdennis) [reply quartz/src/main/java/org/quartz/impl/jdbcjobstore/StdJDBCDelegate.java]: It's not safer - they're exactly the same in terms of safety. If `serializeObject(...)` throws there is nothing to close... and if it doesn't then we close the returned reference. Either way anything created is closed.

File: quartz/src/main/java/org/quartz/plugins/xml/XMLSchedulingDataProcessorPlugin.java (hunk 3 above)
Comment (chrisdennis) [reply quartz/src/main/java/org/quartz/plugins/xml/XMLSchedulingDataProcessorPlugin.java:226]: I'll let @jhouserizer make the call.

File: quartz/src/main/java/org/quartz/simpl/CascadingClassLoadHelper.java (hunk 4 above)
Comment (bcode2) [reply quartz/src/main/java/org/quartz/simpl/CascadingClassLoadHelper.java:110]: Umm I dont get it. The assignment is done just before the break

Comment (chrisdennis) [reply quartz/src/main/java/org/quartz/simpl/CascadingClassLoadHelper.java:110]: More explicitly:
```java
for (ClassLoadHelper helper : loadHelpers) {
    try {
        clazz = helper.loadClass(name);
        loadHelper = helper;
        break;
    } catch (Throwable t) {
        throwable = t;
    }
}
```
but given the wider context you could go to:
```java
public Class<?> loadClass(String name) throws ClassNotFoundException {
    if (bestCandidate != null) {
        try {
            return bestCandidate.loadClass(name);
        } catch (Throwable t) {
            bestCandidate = null;
        }
    }

    Throwable throwable = null;

    for (ClassLoadHelper helper : loadHelpers) {
        try {
            Class<?> clazz = helper.loadClass(name);
            bestCandidate = helper;
            return clazz;
        } catch (Throwable t) {
            throwable = t;
        }
    }

    if (throwable instanceof ClassNotFoundException) {
        throw (ClassNotFoundException) throwable;
    } else {
        throw new ClassNotFoundException( String.format( "Unable to load class %s by any known loaders.", name), throwable);
    } 
}
```

File: quartz/src/main/java/org/quartz/xml/ValidationException.java (hunk 5 above)
Comment (jhouserizer) [reply quartz/src/main/java/org/quartz/xml/ValidationException.java:132]: Agreed, it does look like a bug

**Review by jhouserizer — CHANGES_REQUESTED** [review]

File: quartz/src/main/java/org/quartz/impl/jdbcjobstore/JobStoreSupport.java [hunk 8]
```diff
@@ -1274,33 +1270,29 @@ public boolean removeJobs(final List<JobKey> jobKeys) throws JobPersistenceExcep
...
-                        // FUTURE_TODO: make this more efficient with a true bulk operation...
-                        for (TriggerKey triggerKey : triggerKeys)
-                            allFound = removeTrigger(conn, triggerKey) && allFound;
+                    // FUTURE_TODO: make this more efficient with a true bulk operation...
+                    for (TriggerKey triggerKey : triggerKeys)
+                        allFound = removeTrigger(conn, triggerKey) && allFound;
 
-                        return allFound ? Boolean.TRUE : Boolean.FALSE;
-                    }
+                    return allFound ? Boolean.TRUE : Boolean.FALSE;
```
[the remark this answers is not in the document]
Comment (bcode2) [reply quartz/src/main/java/org/quartz/impl/jdbcjobstore/JobStoreSupport.java:1295]: but with the original  code or current solution  at the end we need to cast  or it wont compile:(

`public boolean removeTriggers(final List<TriggerKey> triggerKeys)
            throws JobPersistenceException {
        return (boolean) executeInLock(
                LOCK_TRIGGER_ACCESS,
                (TransactionCallback) conn -> {
                    boolean allFound = true;

                    // FUTURE_TODO: make this more efficient with a true bulk operation...
                    for (TriggerKey triggerKey : triggerKeys)
                        allFound = removeTrigger(conn, triggerKey) && allFound;

                    return Boolean.valueOf(allFound);
                });
    }`

[the remark this answers is not in the document]
Comment (chrisdennis) [reply quartz/src/main/java/org/quartz/impl/jdbcjobstore/JobStoreSupport.java]: Weird, the above compiles for me, albeit with a warning about an unchecked assignment. More importantly though, if I remove the raw type, and the cast it all works cleanly (no compile errors or warnings):
```java
public boolean removeTriggers(final List<TriggerKey> triggerKeys)
        throws JobPersistenceException {
    return executeInLock(
            LOCK_TRIGGER_ACCESS,
            conn -> {
                boolean allFound = true;

                // FUTURE_TODO: make this more efficient with a true bulk operation...
                for (TriggerKey triggerKey : triggerKeys)
                    allFound &= removeTrigger(conn, triggerKey);

                return allFound;
            });
}
```

**Comment by jhouserizer** [comment]
Can you help me understand where this all stands at this point?

**Comment by bcode2** [comment]
> Can you help me understand where this all stands at this point?

Hi @jhouserizer  give me a second . I guess I have already addressed all comments. Will rollback one  of those that are still unfixed to allow you to merge it. In another smaller PR will try to fix those referenced  by @chrisdennis  and @melloware

**Comment by bcode2** [comment]
Well @jhouserizer , @chrisdennis I guess  everything has been solved

**Review by jhouserizer — APPROVED** [review]

**Pull request merged**